ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/InternalCustomerCenter.kt

```diff
@@ -120,6 +169,143 @@ private fun CustomerCenterLoaded(
+                        Text(
+                            text = "Going to check for previous purchases",
+                            style = MaterialTheme.typography.bodyLarge,
+                        )
+                    },
+                    confirmButton = {
+                        Button(
+                            onClick = {
+                                isRestoring = true
+                                onRestore()
+                            },
+                        ) {
+                            Text("Check Past Purchases")
+                        }
+                    },
+                    dismissButton = {
+                        TextButton(
+                            onClick = {
+                                isRestoring = false
+                                onDismiss()
+                            },
+                        ) {
+                            Text("Cancel")
+                        }
+                    },
+                )
+            }
+        }
+        RestorePurchasesState.PurchasesRecovered -> {
+            AlertDialog(
+                onDismissRequest = onDismiss,
+                title = {
+                    Text(
+                        text = "Purchases Recovered",
+                        style = MaterialTheme.typography.headlineSmall,
+                    )
+                },
+                text = {
+                    Text(
+                        text = "Your purchases have been restored successfully",
+                        style = MaterialTheme.typography.bodyLarge,
+                    )
+                },
+                confirmButton = {
+                    Button(onClick = onDismiss) {
+                        Text("Dismiss")
+                    }
+                },
+            )
+        }
+        RestorePurchasesState.PurchasesNotFound -> {
+            AlertDialog(
+                onDismissRequest = onDismiss,
+                title = {
+                    Text(
+                        text = "No Purchases Found",
+                        style = MaterialTheme.typography.headlineSmall,
+                    )
+                },
+                text = {
+                    Text(
+                        text = "No previous purchases were found",
+                        style = MaterialTheme.typography.bodyLarge,
+                    )
+                },
+                confirmButton = {
+                    Button(onClick = onContactSupport) {
+                        Text("Contact Support")
+                    }
+                },
+                dismissButton = {
+                    Button(onClick = onDismiss) {
+                        Text("Dismiss")
+                    }
+                },
+            )
+        }
+    }
+}
+
+sealed class RestorePurchasesState {
+    object Initial : RestorePurchasesState()
+    object PurchasesRecovered : RestorePurchasesState()
+    object PurchasesNotFound : RestorePurchasesState()
```

Comment:
Hmm could this be a normal enum? (not sure if we're planning to add additional state in any of these though)

Reply:
Sure, I had one with state that ended up removing so it doesn't make sense anymore